simplecpp.cpp

```diff
@@ -2341,6 +2341,63 @@ static void simplifySizeof(simplecpp::TokenList &expr, const std::map<std::strin
     }
 }
 
+/** Evaluate __has_include(file) */
+static std::string openHeader(std::ifstream &f, const simplecpp::DUI &dui, const std::string &sourcefile, const std::string &header, bool systemheader);
+static void simplifyHasInclude(simplecpp::TokenList &expr, const simplecpp::DUI &dui)
+{
+    for (simplecpp::Token *tok = expr.front(); tok; tok = tok->next) {
```

Comment:
I think it would feel better if there was some testing of this. could we refactor and break up this perhaps so during testing we test if the filename argument for different `__has_include` are extracted properly. i.e. `__has_include("foo.h")` => `"foo.h"`. `__has_include(<path/foo.h>)` => `<path/foo.h>`. ?

Reply:
It is my fault there are not tests, but that is making me worried when I review this.

Reply:
I added several tests here: https://github.com/al42and/simplecpp/commit/c653ae8cf5e4e0f3d20235654b8d6cf1240ac852

Not nearly exhaustive enough, but covers most common cases, IMO.

Please tell me whether you want me to make a PR or whatever.

Reply:
Hi Andrey,

I haven't found the time to finish this in the previous 6 months and it's unlikely that this will change in short term.

Feel free to take over this branch and finish it.

regards
Christian